src/fetch_queries.py

```diff
@@ -1,52 +1,62 @@
+"""
+Fetch queries for products in a specific category
+"""
+
+import argparse
+import asyncio
 import json
+import logging
 import os
-import argparse
+
+import aiofiles
 import pandas as pd
-from tqdm import tqdm
-from product_categories import ProductCategory
+from tqdm.asyncio import tqdm
 
+from product_config import ProductCategory
 
-def load_valid_asins(category: str) -> list:
+logging.basicConfig(
+    level=logging.INFO, format="%(asctime)s - %(levelname)s - %(message)s"
+)
+
+
+async def load_valid_asins(category: str) -> list:
     """Load valid ASINs from the output directory"""
     category_filename = category.lower().replace(" & ", "_").replace(" ", "_")
     asins_file = f"output/{category_filename}_asins.json"
 
     try:
-        with open(asins_file, "r") as f:
-            return json.load(f)
+        async with aiofiles.open(asins_file, "r", encoding="utf-8") as f:
+            return json.loads(await f.read())
     except FileNotFoundError:
-        print(f"No valid ASINs file found at {asins_file}")
-        print("Please run validate-product-images.py first!")
+        logging.error("No valid ASINs file found at %s", asins_file)
+        logging.error("Please run validate-product-images.py first!")
         return []
 
 
-def fetch_category_queries(
-    category: str, valid_asins: list, locale: str, labels: list
-) -> list:
+async def fetch_valid_queries(valid_asins: list, locale: str, labels: list) -> list:
     """Fetch queries related to valid ASINs for the given category"""
-    base_path = "../esci-data/shopping_queries_dataset"
+    base_path = "esci-data/shopping_queries_dataset"
 
-    print("Reading parquet files...")
+    logging.info("Reading parquet files...")
     try:
         df_examples = pd.read_parquet(
             f"{base_path}/shopping_queries_dataset_examples.parquet"
         )
-        df_products = pd.read_parquet(
-            f"{base_path}/shopping_queries_dataset_products.parquet"
-        )
     except Exception as e:
-        print(f"Error reading parquet files: {e}")
+        logging.error("Error reading parquet files: %s", e)
         return []
 
-    print("Filtering products...")
+    logging.info("Filtering products...")
     # Filter products by valid ASINs, locale and labels
     df_examples_filtered = df_examples[
         (df_examples["product_id"].isin(valid_asins))
         & (df_examples["product_locale"] == locale)
         & (df_examples["esci_label"].isin(labels))
+        & (df_examples["split"] == "train")
```

Comment:
@git-yogesh why are we only considering train splits?

Reply:
This is because we can potentially fine-tune models on "train" set later, and use "test" set for predictions. It does not have a significance at the moment.